actors/market/src/lib.rs

```diff
@@ -429,63 +392,77 @@ impl Actor {
 
         let mut new_deal_ids = Vec::with_capacity(valid_deals.len());
         rt.transaction(|st: &mut State, rt| {
-            let mut msm = st.mutator(rt.store());
-            msm.with_pending_proposals(Permission::Write)
-                .with_deal_proposals(Permission::Write)
-                .with_deals_by_epoch(Permission::Write)
-                .with_escrow_table(Permission::Write)
-                .with_locked_table(Permission::Write)
-                .build()
-                .map_err(|e| {
-                    e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to load state")
-                })?;
+            let mut pending_deals: Vec<Cid> = vec![];
+            let mut deal_proposals: Vec<(DealID, DealProposal)> = vec![];
+            let mut deals_by_epoch: Vec<(ChainEpoch, DealID)> = vec![];
+            let mut pending_deal_allocation_ids: Vec<(BytesKey, AllocationID)> = vec![];
+
             // All storage dealProposals will be added in an atomic transaction; this operation will be unrolled if any of them fails.
             // This should only fail on programmer error because all expected invalid conditions should be filtered in the first set of checks.
             for valid_deal in valid_deals.iter() {
-                msm.lock_client_and_provider_balances(&valid_deal.proposal)?;
+                st.lock_client_and_provider_balances(rt.store(), &valid_deal.proposal).map_err(
+                    |e| {
+                        e.downcast_default(
+                            ExitCode::USR_ILLEGAL_STATE,
+                            "failed to check for existence of deal proposal",
+                        )
+                    },
+                )?;
```

Comment:
Please push basic error mapping down into the state layer. All the state functions should return `Result<T, ActorError>` so that here we can just use `?`. `anyhow` should not escape the state layer.

Reply:
Done 👍🏾